Add HPA

zwolf opened the pull request:
Max on this one is 4 since there's 3 running right now. If you feel like that's already too many, I can drop the max.

zwolf committed:
Add HPA

camallen commented on kubernetes/deployment.tmpl:
```diff
@@ -26,3 +26,16 @@ spec:
+    kind: Deployment
+    name: thumbnailer
+  minReplicas: 1
+  maxReplicas: 3
```
I'd let this scale up if we get busy. Cpu / ram use is low in this pod so shouldn't put much pressure on the nodes / cluster.
```suggestion
  maxReplicas: 6
```

camallen committed:
remove deployment replicas directive

rely on the HPA replica directives

camallen committed:
increase cpu request

target based horizontal pod autoscalers rely on the resource request for the percentage of use metric, thus previously this was using (avg sum CPU)/150Mi to calc the hpa cpu metric for scaling pods. Thus the HPA would scale the pods based on low cpu request metric

See https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/

camallen reviewed (approved):
LGTM - i'd probably bump the max number of these we can run as if we get busy this app serves a bit of traffic.

zwolf committed:
More max replicas

Co-authored-by: Campbell Allen <campbell.allen@gmail.com>
